package/deploy_package_index.sh

```diff
@@ -1,32 +1,35 @@
 #!/bin/bash
-#
 # This script updates package index hosted on esp8266.github.io (aka arduino.esp8266.com).
-# Normally is run by Travis CI for tagged versions, as a deploy step.
 
 tag=`git describe --tags`
 
 cd $(dirname "$0")
 
+set -e # Abort with error if anything here does not go as expected!
+
 # Decrypt and install SSH private key.
 # "encrypted_xxx_key" and "encrypted_xxx_iv" are environment variables
 # known to Travis CI builds.
-openssl aes-256-cbc -K $encrypted_3a94a4db7dec_key -iv $encrypted_3a94a4db7dec_iv -in esp8266_github_io_deploy.enc -out esp8266_github_io_deploy -d
+openssl enc -d -aes-256-cbc -K $GHKEY -iv $GHIV -in esp8266_github_io_deploy.enc -out esp8266_github_io_deploy
 eval "$(ssh-agent -s)"
 chmod 600 esp8266_github_io_deploy
 ssh-add esp8266_github_io_deploy
+mkdir -p ~/.ssh
+chmod go-w ~/.ssh
 echo -e "Host github.com\nStrictHostKeyChecking no\n" >> ~/.ssh/config
+chmod go-w  ~/.ssh/config
 
 # Clone the Github pages repository
 git clone git@github.com:esp8266/esp8266.github.io.git
 pushd esp8266.github.io
 
 # Update the package index
-cp ../versions/$tag/package_esp8266com_index.json stable/package_esp8266com_index.json
+cp ../versions/*/package_esp8266com_index.json stable/package_esp8266com_index.json
```

Comment:
I'm pretty sure and it is tested.  The Travis tag and the GH tag are not identical, so the subdir can and will be different and it's either add some powershell code here to regen, or stick with wildcard.

In CI we was guaranteed to start with a clean FS and the script only ever generates a single subdir, so this is safe.  Also, if something really bad happens and we have 2 different subdirs, this will report failure and not upload anything (multiple files copy destination is not a subdir).